envoy/api/v2/core/health_check.proto

```diff
@@ -116,6 +117,16 @@ message HealthCheck {
     string service_name = 1;
   }
 
+  // [#not-implemented-hide:] Custom health check.
+  message ExtensionHealthCheck {
```

Comment:
I kind of think something like `HealthCheckPlugin`, `CustomHealthCheck` etc. might be better than `ExtensionHealthCheck` as a name here.

Reply:
OK, I think `CustomHealthCheck` is fine. Will wait for other suggestions.

Reply:
`CustomHealthCheck` SGTM.